pkgs/applications/editors/emacs/elisp-packages/updater-emacs.nix

```diff
@@ -1,33 +1,8 @@
 let
   pkgs = import ../../../../.. {};
 
-  emacsEnv = pkgs.emacs.pkgs.withPackages (epkgs: let
-
-    promise = epkgs.trivialBuild {
-      pname = "promise";
-      version = "1";
-      src = pkgs.fetchFromGitHub {
-        owner = "bendlas";
-        repo = "emacs-promise";
-        rev = "4da97087c5babbd8429b5ce62a8323b9b03c6022";
-        sha256 = "0yin7kj69g4zxs30pvk47cnfygxlaw7jc7chr3b36lz51yqczjsy";
-      };
-    };
-
-    semaphore = epkgs.trivialBuild {
-      pname = "semaphore";
-      version = "1";
-      packageRequires = [ promise ];
-      src = pkgs.fetchFromGitHub {
-        owner = "webnf";
-        repo = "semaphore.el";
-        rev = "93802cb093073bc6a6ccd797328dafffcef248e0";
-        sha256 = "09pfyp27m35sv340xarhld7xx2vv5fs5xj4418709iw6l6hpk853";
-      };
-    };
-
-  in [ promise semaphore ]);
-
+  emacsEnv = pkgs.emacs.pkgs.withPackages(epkgs: with epkgs; [
+    promise semaphore ]);
```

Comment:
Does the manual update script still work after this change?  I made this change a few months ago and found that this change broke the manual update script.

I am not sure if this change is desirable because pinning some dependencies makes the manual update script more reproducible.

Reply:
Oh, just find that you add the pinned version of `async-await`, `promise` and `semaphore` to manual packages, which shadows newer version of them on MELPA.  I do not think that is a good idea.

Reply:
They are on melpa??
I thougth they weren't...

Why pinning them? Those packages are so stable they have no changes for years...

Nonetheless it looks cleaner to melpaize them...

Reply:
> Why pinning them?

There were a few patches not in the upstream promise packages back then.

Reply:
For now I will merely unpin them.